skbio/tree/_tree.py

```diff
@@ -211,6 +215,9 @@ def _copy(self, deep, memo):
         if hasattr((root := self.root()), "_registered_caches"):
             exclude_attrs = exclude_attrs | root._registered_caches
 
+        # exclude dynamically generated methods
+        exclude_attrs = exclude_attrs | {"_write_method"}
```

Comment:
This edit rings an alarm to me. I am not sure if this will work or not. But can you try to make a `TreeNode` object, copy it, and see if the copy still has the `write` method?